odoo/addons/base/static/src/js/res_config_settings.js

```diff
@@ -344,22 +347,23 @@ var BaseSettingRenderer = FormRenderer.extend({
             this._resetSearch();
         }
     },
+
     /**
      * highlight search word
      *
      * @private
-     * @param {string} text
-     * @param {string} word
+     * @param {HTMLElement} node
+     * @param {string} upperCasedSearchText
      */
-    _wordHighlighter: function (text, word) {
-        if (text.indexOf('highlighter') !== -1) {
-            text = text.replace('<span class="highlighter">', "");
-            text = text.replace("</span>", "");
-        }
-        var match = text.search(new RegExp(word, "i"));
-        word = text.substring(match, match + word.length);
-        var highlightedWord = "<span class='highlighter'>" + word + '</span>';
-        return text.replace(word, highlightedWord);
+    _wordHighlighter: function (node, upperCasedSearchText) {
+        const text = node.textContent;
+        const startIndex = text.toUpperCase().indexOf(upperCasedSearchText);
+        const endIndex = startIndex + upperCasedSearchText.length;
+        $(node).empty().append(
+            text.substring(0, startIndex),
```

Comment:
As you noted elsewhere, this is broken because jquery will parse that text as HTML :/ (same for the trailing edge)

Needs to be something along the lines of `document.createTextNode(...)` or possibly just `new Text(...)` if we consider that that's well enough supported.